src/Domain/CoordinateMaps/CoordinateMap.hpp

```diff
@@ -267,11 +356,31 @@ template <typename T>
 constexpr SPECTRE_ALWAYS_INLINE tnsr::I<
     T, CoordinateMap<SourceFrame, TargetFrame, Maps...>::dim, TargetFrame>
 CoordinateMap<SourceFrame, TargetFrame, Maps...>::call_impl(
-    tnsr::I<T, dim, SourceFrame>&& source_point) const {
+    tnsr::I<T, dim, SourceFrame>&& source_point, const double time,
+    const std::unordered_map<std::string, FunctionOfTime&>& f_of_t_list) const
+    noexcept {
   std::array<T, dim> mapped_point = make_array<T, dim>(std::move(source_point));
-  tuple_fold(maps_, [](const auto& map,
-                       std::array<T, dim>& point) { point = map(point); },
-             mapped_point);
+  tuple_fold(
+      maps_,
+      // chooses the correct call based on time-dependence of map
+      make_overloader(
+          [](const auto& the_map, std::array<T, dim>& point, const double /*t*/,
```

Comment:
I think it's much cleaner and easier to understand to take `true_type` and `false_type` and overload by that as opposed to use SFINAE with Requires. There are 3 other blocks where this should also be done.